services/api-server/tests/unit/conftest.py

```diff
@@ -165,21 +171,24 @@ def make_engine(postgres_service: Dict) -> Callable:
     dsn = postgres_service["dsn"]  # session scope freezes dsn
 
     def maker(is_async=True) -> Union[Coroutine, Callable]:
-        return aiopg.sa.create_engine(dsn) if is_async else sa.create_engine(dsn)
+        if is_async:
+            return aiopg.sa.create_engine(dsn)
+        return sa.create_engine(dsn)
 
     return maker
 
 
 @pytest.fixture
-def apply_migration(postgres_service: Dict, make_engine) -> None:
+def apply_migration(postgres_service: Dict, make_engine) -> Iterator[None]:
     kwargs = postgres_service.copy()
     kwargs.pop("dsn")
     pg_cli.discover.callback(**kwargs)
     pg_cli.upgrade.callback("head")
+
     yield
+
     pg_cli.downgrade.callback("base")
     pg_cli.clean.callback()
-
     # FIXME: deletes all because downgrade is not reliable!
     engine = make_engine(False)
     metadata.drop_all(engine)
```

Comment:
why not using pytest simcore fixtures for the database?

Reply:
the ``postgres_db`` fixture is indeed equivalent to ``apply_migration`` but they have different scope. The first is at module scope and the latter i need it at function scope